addons/pos_self_order/controllers/orders.py

```diff
@@ -8,26 +8,24 @@
 from werkzeug.exceptions import NotFound, BadRequest, Unauthorized
 
 class PosSelfOrderController(http.Controller):
-    @http.route("/pos-self-order/process-new-order", auth="public", type="json", website=True)
-    def process_new_order(self, order, access_token, table_identifier):
+    @http.route("/pos-self-order/process-new-order/<device_type>/", auth="public", type="json", website=True)
+    def process_new_order(self, order, access_token, table_identifier, device_type):
         lines = order.get('lines')
         pos_config, table = self._verify_authorization(access_token, table_identifier)
         pos_session = pos_config.current_session_id
 
-        table_or_config_id = table.id if table else pos_config.id
-        sequence_number = self._get_sequence_number(table_or_config_id, pos_session.id)
-        unique_id = self._generate_unique_id(pos_session.id, table_or_config_id, sequence_number)
-
+        ir_sequence = pos_config.env['ir.sequence'].with_context(company_id=pos_config.company_id.id).next_by_code('pos.session')
```

Comment:
This sequence is the sequence of the pos.session. We should absolutely not use this sequence for the orders.